oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/facets/ElasticStatisticalFacetAsyncProvider.java

```diff
@@ -74,11 +76,15 @@ public class ElasticStatisticalFacetAsyncProvider implements ElasticFacetProvide
 
         this.elasticResponseHandler = elasticResponseHandler;
         this.isAccessible = isAccessible;
-        this.facetFields = elasticRequestHandler.facetFields().collect(Collectors.toSet());
+        this.facetFields = elasticRequestHandler.facetFields().
+                collect(Collectors.toSet());
+        Set<String> elasticFieldNames = facetFields.stream().
```

Comment:
Same comment made in `ElasticSecureFacetAsyncProvider`. We don't need to keep both Sets. Only the one with the fields converted using `ElasticIndexUtils ` is required.

Reply:
I see! I'll try to simplify this.